app/github.server.ts

```diff
@@ -75,3 +76,139 @@ export async function getPullRequestInfo(
 
   return value.repository.pullRequest;
 }
+
+const TARGET_OWNER = "Codesee-io";
+const TARGET_REPO = "opensourcehub";
+
+export type FileToUpload = {
+  content: string;
+  filePath: string;
+  encoding: "utf-8" | "base64";
+};
+
+/**
+ * Create a new branch with a new commit that contains one or more files, and
+ * then open a pull request as the current user.
+ *
+ * I want to thank the GitHub API for making this needlessly difficult.
+ */
+export async function createNewPullRequest(
```

Comment:
Now, we're ready to upload our files (which include an MDX file and maybe an image for the avatar).

I spent half a day figuring out the correct GitHub incantations to make this work. Essentially:
1. we upload our new files to GitHub
2. we create a new Git tree in order to place the new files in the right location
3. we create a new GitHub branch
4. we create a new commit on that new branch
5. we open a pull request against OSH

Reply:
**Praise**: This looks great. Thank you for figuring this out. Either Robin and I missed the Git database API when we were trying to get multi-file commits to work, or it didn't exist at the time. Either way, kudos to you!